### DISCUSSION TITLE
STM32h753zit6 Custom board, uploads but crashes

### DISCUSSION THREAD

**Discussion by FSkavlem** [stm32duino/Arduino_Core_STM32#2532] [Q&A] [answered]
Currently debugging a prototype i am building, its a custom PCB built with STM32h753zit6

Here is the case, i have used the nucleo H753ZI as reference when laying out the PCB, the custom board works fine when loading with STM32 cubeIDE.

When loading with arduino IDE, the program uploads but the MCU immediately halts. 
When loading the same program with arduino IDE to the nucleo H753ZI it runs fine. 
(both loaded with same external STLINK)

The only difference between these 2 cards at this moment is that on the nucleo the MCU installed is STM32h753ZIT6U(these i cannot purchase) and on the custom PCB the MCU installed is STM32h753zit6

Any particular reason it should work on a STM32h753ZIT6U but not a STM32h753zit6?

I need some fresh ideas to continue debugging, any ideas?

**Comment by fpistm** [comment]
The Nucleo uses the HSE clock. Do you have a HSE clock on your custom PCB ?

**Reply by FSkavlem** [reply]
the design is open for both so tried with external 25MHZ HSE clock and 32.768khz LSE clock attached

**Reply by FSkavlem** [reply]
my nucleo-H753ZI does not have a HSE clock, its not mounted. Only has 32.768khz LSE

**Reply by FSkavlem** [reply, marked as answer]
!"¤%" you are right, the nucleo gets a HSE 8 MHZ from the stlink STLK MCO, my design got a 25 Mhz. Ill order some new ones and change it for 8 Mhz. Thanks for a point in the right direction!

**Comment by FSkavlem** [comment]
I can confirm it now works. This whole issue occured because of the reference design for the STM32H753ZIT6 there is a 25.000 Mhz clock on HSE. The nucleo h753zi has no external ceramic resonator mounted but receive a 8.000 Mhz CLK from the onboard ST link. In order to make your own board you must add a external 8.000 Mhz clock

**Reply by fpistm** [reply]
Or provide correct core clock config :wink:

**Comment by FSkavlem** [comment]
I just want to add an update on this, installed the new 8 mhz clock and it still dont run, i guess its because its setup to receive a 8 mhz clock. They way you can make any clock to work with arduino core on stm32 is to open stm32cubeIDE and setup the selected chip in the .ioc  with the right clock setup there, then copy the void SystemClock_Config(void) function from cubeide into your arduino code and call the function first line in void setup() works like a charm, both with 25 Mhz HSE and 8 Mhz Hse guess it will work with any clock configurations inside specs for the selected chip.

**Reply by fpistm** [reply]
Not fully true. By default an `HSE_VALUE` is defined. So if the one you used is different you also have to redefine it as it is used for some computation.
Here the default value for H7:
https://github.com/stm32duino/Arduino_Core_STM32/blob/42e02623a246f06fb5748b362c2087d04bb37c0e/system/STM32H7xx/stm32h7xx_hal_conf_default.h#L110-L112

You can create an `hal_conf_extra.h` to define the correct value:
https://github.com/stm32duino/Arduino_Core_STM32/wiki/HAL-configuration#customize-hal-or-variant-definition
